tests/UI/campaigns/regression/currencies/computingPrecision_FO.js

```diff
@@ -0,0 +1,535 @@
+
+        const validationMessage = await addCartRulePage.createEditCartRules(page, percentCartRule);
+        await expect(validationMessage).to.contains(addCartRulePage.successfulCreationMessage);
+      });
+    });
+
+    describe('Create a gift cart rule', async () => {
+      it('should go to new cart rule page', async function () {
+        await testContext.addContextItem(this, 'testIdentifier', 'goToNewCartRulePage2', baseContext);
+
+        await cartRulesPage.goToAddNewCartRulesPage(page);
+        const pageTitle = await addCartRulePage.getPageTitle(page);
+        await expect(pageTitle).to.contains(addCartRulePage.pageTitle);
+      });
+
+      it('should create new cart rule', async function () {
+        await testContext.addContextItem(this, 'testIdentifier', 'createGiftCartRule', baseContext);
+
+        const validationMessage = await addCartRulePage.createEditCartRules(page, giftCartRule);
+        await expect(validationMessage).to.contains(addCartRulePage.successfulCreationMessage);
+      });
+    });
+  });
+
+  describe('Place an order with discounts in FO', async () => {
+    it('should go to FO page', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'viewMyShop', baseContext);
+
+      page = await cartRulesPage.viewMyShop(page);
+      await homePage.changeLanguage(page, 'en');
+
+      const isHomePage = await homePage.isHomePage(page);
+      await expect(isHomePage, 'Fail to open FO home page').to.be.true;
+    });
+
+    it('should go to login page', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'goToFOLoginPage', baseContext);
+
+      await homePage.goToLoginPage(page);
+      const pageTitle = await foLoginPage.getPageTitle(page);
+      await expect(pageTitle, 'Fail to open FO login page').to.contains(foLoginPage.pageTitle);
+    });
+
+    it('should sign in with default customer', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'loginInFO', baseContext);
+
+      await foLoginPage.customerLogin(page, DefaultAccount);
+      const isCustomerConnected = await foLoginPage.isCustomerConnected(page);
+      await expect(isCustomerConnected, 'Customer is not connected').to.be.true;
+    });
+
+    it('should add product to cart', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'addProductToCart', baseContext);
+
+      // Go to home page
+      await foLoginPage.goToHomePage(page);
+
+      // Go to product page after searching its name
+      await homePage.searchProduct(page, orderToMake.product.name);
+      await searchResultsPage.goToProductPage(page, 1);
+
+      // Add the created product to the cart
+      await productPage.addProductToTheCart(page, orderToMake.quantityToOrder);
+
+      // Check cart page
+      const isCartPage = await cartPage.isCartPage(page);
+      await expect(isCartPage, 'Fail to go to cart page').to.be.true;
```

Comment:
What is this method `isCartPage` ? Is this specific to the `cartPage` class ? Is there a reason you didn't use the title check like we do for all the other pages (genuine question) ?

Reply:
Should be enough to check page title (I guess I was overthinking)